### PR TITLE
main postgres 9.5 installation routine

### PR DESCRIPTION
**Pull request by jpeerz**
tested as next using puppet apply:

```
root@vagrant-ubuntu-trusty-64:/home/vagrant# ls -l /etc/postgresql/9.5/main/
environment      pg_ctl.conf      pg_hba.conf      pg_ident.conf    postgresql.conf  start.conf       
root@vagrant-ubuntu-trusty-64:/home/vagrant# puppet apply --modulepath=/vagrant/puppet/modules/ /vagrant/puppet/manifests/orcid_jenkins.pp 
Notice: Compiled catalog for vagrant-ubuntu-trusty-64.avantica.avanticatec.net in environment production in 0.44 seconds
Notice: Installing Postgres...
Notice: /Stage[main]/Orcid_base::Postgresql_95/Notify[Installing Postgres...]/message: defined 'message' as 'Installing Postgres...'
Notice: /Stage[main]/Orcid_base::Postgresql_95/Exec[pg_install_apt_key]/returns: executed successfully
Notice: /Stage[main]/Orcid_base::Postgresql_95/File[/etc/apt/sources.list.d/pgdg.list]/content: content changed '{md5}f8cac3af96032517017fc7185a045987' to '{md5}e8545ab42b563cc1654788e3adad96a0'
Notice: /Stage[main]/Orcid_base::Postgresql_95/Exec[pg_upgate_codename]/returns: executed successfully
Notice: /Stage[main]/Orcid_base::Postgresql_95/Exec[pg_upgate_ubuntu_packages_db]/returns: executed successfully
Notice: /Stage[main]/Orcid_base::Postgresql_95/File[/etc/postgresql/9.5/main/pg_hba.conf]/content: content changed '{md5}cbf62fe357451a5b84acf6e43e82329f' to '{md5}9da74fb72f8c8de2d28b69f1f48ad7dc'
Notice: /Stage[main]/Orcid_base::Postgresql_95/File[/etc/postgresql/9.5/main/pg_hba.conf]/mode: mode changed '0640' to '0440'
Notice: Finished catalog run in 17.14 seconds
root@vagrant-ubuntu-trusty-64:/home/vagrant# ls -l /etc/postgresql/9.5/main/
total 52
-rw-r--r-- 1 postgres postgres   315 Oct 12 19:16 environment
-rw-r--r-- 1 postgres postgres   143 Oct 12 19:16 pg_ctl.conf
-r--r----- 1 postgres postgres   389 Oct 12 19:31 pg_hba.conf
-rw-r----- 1 postgres postgres  4641 Oct 12 19:16 pg_hba.conf.puppet-bak
-rw-r----- 1 postgres postgres  1636 Oct 12 19:16 pg_ident.conf
-rw-r--r-- 1 postgres postgres 21448 Oct 12 19:16 postgresql.conf
-rw-r--r-- 1 postgres postgres   317 Oct 12 19:16 start.conf
root@vagrant-ubuntu-trusty-64:/home/vagrant#
```

### CODE REVIEW

File: puppet/modules/orcid_base/files/etc_postgresql_95_pg_hba.conf [hunk 1]
```diff
@@ -0,0 +1,8 @@
+local   all           postgres    ident
+local   replication   postgres    ident
+local   all           all                       md5
+host    all           all         127.0.0.1/32  md5
+host    all           all         ::1/128       md5
+host    all	          all		      10.0.0.0/8		md5
```
Comment (rcpeters) [inline puppet/modules/orcid_base/files/etc_postgresql_95_pg_hba.conf]: configure this to be private

Comment (jpeerz) [inline puppet/modules/orcid_base/files/etc_postgresql_95_pg_hba.conf]: by private you mean, to only accept local connections ?

Comment (rcpeters) [inline puppet/modules/orcid_base/files/etc_postgresql_95_pg_hba.conf]: I mean not share the public configuration with everyone. I.E. put in a variable.

**Comment by rcpeters** [comment]
Let's also add 
/etc/postgresql/9.5/main/postgresql.conf with a variable that has our settings at the end of the file (you'll see us using it in prod).

**Comment by jpeerz** [comment]
second execution won't take effect

```
root@vagrant-ubuntu-trusty-64:~# puppet apply --modulepath=/vagrant/puppet/modules/ /vagrant/puppet/manifests/orcid_db.pp 
Notice: Compiled catalog for vagrant-ubuntu-trusty-64 in environment production in 0.39 seconds
Notice: /Stage[main]/Orcid_base::Postgresql/File[/etc/apt/sources.list.d/pgdg.list]/ensure: defined content as '{md5}e8545ab42b563cc1654788e3adad96a0'
Notice: /Stage[main]/Orcid_base::Postgresql/Exec[pg_update_codename]/returns: executed successfully
Notice: /Stage[main]/Orcid_base::Postgresql/Package[postgresql-9.5]/ensure: ensure changed 'purged' to 'present'
Notice: /Stage[main]/Orcid_base::Postgresql/File[/etc/postgresql/9.5/main/postgresql.conf]/content: content changed '{md5}e3bcb5580221e4dd1fe47163b0bde570' to '{md5}647fdd8ef4b85465dfe38a856dbf07bd'
Notice: /Stage[main]/Orcid_base::Postgresql/File[/etc/postgresql/9.5/main/postgresql.conf]/mode: mode changed '0644' to '0440'
Notice: /Stage[main]/Orcid_base::Postgresql/File[/etc/postgresql/9.5/main/pg_hba.conf]/content: content changed '{md5}cbf62fe357451a5b84acf6e43e82329f' to '{md5}37f405b467b2ce39caa3ca14a933bc8b'
Notice: /Stage[main]/Orcid_base::Postgresql/File[/etc/postgresql/9.5/main/pg_hba.conf]/mode: mode changed '0640' to '0440'
Notice: Finished catalog run in 15.58 seconds
root@vagrant-ubuntu-trusty-64:~# cat /etc/postgresql/9.5/main/pg_hba.conf

local   all             postgres                                peer
local   all             all                                     peer
host    all             all             127.0.0.1/32            md5
host    all             all             ::1/128                 md5
root@vagrant-ubuntu-trusty-64:~# 
root@vagrant-ubuntu-trusty-64:~#                                                                                                                                                                                   
root@vagrant-ubuntu-trusty-64:~# puppet apply --modulepath=/vagrant/puppet/modules/ /vagrant/puppet/manifests/orcid_db.pp                                                                                          
Notice: Compiled catalog for vagrant-ubuntu-trusty-64 in environment production in 0.38 seconds                                                                                           
Notice: Finished catalog run in 1.36 seconds                                                                                                                                                                       
root@vagrant-ubuntu-trusty-64:~#  
```

**Pull request merged**